roles/kubernetes-apps/ansible/defaults/main.yml

```
@@ -13,6 +13,22 @@ dns_prevent_single_point_failure: "{{ 'true' if dns_min_replicas | int > 1 else
 enable_coredns_reverse_dns_lookups: true
 coredns_ordinal_suffix: ""
 # dns_extra_tolerations: [{effect: NoSchedule, operator: "Exists"}]
+coredns_affinity:
+  podAntiAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      - topologyKey: "kubernetes.io/hostname"
+        labelSelector:
+          matchLabels:
+            k8s-app: kube-dns{{ coredns_ordinal_suffix }}
+  nodeAffinity:
+    preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        preference:
+          matchExpressions:
+            - key: node-role.kubernetes.io/control-plane
+              operator: In
+              values:
+                - ""
```

This is the upstream default value.

FYI: https://github.com/coredns/helm/blob/1921572adca00de6a8bae2485ddfae0d3d1db0a7/charts/coredns/values.yaml#L193

```suggestion
coredns_affinity: {}
```

---

https://github.com/kubernetes/kubernetes/blob/6e6e6f6e53f22c57793bf0b1cfabe92a7939b6fc/cluster/addons/dns/coredns/coredns.yaml.base#L118-L128

I would keep the pod antiaffinity (as default) for the default.
The nodeAffinity to control plane is something from kubeadm, I think ? It's not in k/k and I think there is something related to cluster init :thinking:  not sure if it should stay as a default, it can cause trouble (such as : https://render.com/blog/a-hidden-dns-dependency-in-kubernetes )

---

Yeah, you're right the manifests embedded in kubeadm don't have that.
Lets go with the podAntiAffinity only for default.

---

```suggestion
coredns_affinity:
  podAntiAffinity:
    preferredDuringSchedulingIgnoredDuringExecution:
    - weight: 100
      podAffinityTerm:
        labelSelector:
          matchExpressions:
            - key: k8s-app
              operator: In
              values: ["kube-dns"]
        topologyKey: kubernetes.io/hostname
```
And the setup is entirely different when I look at it. (face palm laugh)

---

Confirmed!